PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/coreTools/FunctionsCoreToolsManager.kt

```diff
@@ -115,16 +115,24 @@ object FunctionsCoreToolsManager {
 
                 val releaseCoreTools = release.coreTools
                         .filter {
-                            // Match OS and architecture
+                            // Match OS and ensure a download link is present
                             it.os.equals(azureCoreToolsFeedReleaseFilter.os, ignoreCase = true) &&
-                                    it.architecture.equals(azureCoreToolsFeedReleaseFilter.architecture, ignoreCase = true) &&
                                     it.downloadLink.isNotNullOrEmpty
                         }
-                        .minByOrNull {
-                            // Ideally match requested size, if not, fall back to non-matching.
-                            // This is the case in the feed for e.g. v2, where Windows has only full size
-                            if (it.size.equals(azureCoreToolsFeedReleaseFilter.size, ignoreCase = true)) 0 else 1
-                        } ?: continue
+                        .sortedWith(compareBy<ReleaseCoreTool> {
+                            // Then match architecture. Items higher in the list have precedence.
+                            // Use 9999 when no match is found.
+                            azureCoreToolsFeedReleaseFilter.architectures.indexOfFirst { architecture ->
+                                it.architecture.equals(architecture, ignoreCase = true)
+                            }.let { rank -> if (rank >= 0) rank else 9999 }
```

Comment:
Here we have items without a rank moved to the top, if I understand it correctly. Is it what we expect?

Reply:
This is indeed expected.

Reply:
If the requirement is:
* Windows - x64 - minified -> rank 0
* Windows - x64 - full -> rank 1

Then:
* Windows - x86 - full -> rank 9999

Using `sortedBy()`, that gives rank 0 highest priority (and 9999 lowest)